src/bin/src/systems/keep_alive_system.rs

```diff
@@ -46,19 +48,17 @@ impl System for KeepAliveSystem {
                 last_time = current_time;
             }
 
-            let fifteen_seconds_ms = 15000; // 15 seconds in milliseconds
-
             let entities = state
                 .universe
-                .query::<(&mut StreamWriter, &ConnectionState, &KeepAlive)>()
+                .query::<(&mut StreamWriter, &ConnectionState)>()
                 .into_entities()
                 .into_iter()
                 .filter_map(|entity| {
                     let conn_state = state.universe.get::<ConnectionState>(entity).ok()?;
-                    let keep_alive = state.universe.get_mut::<KeepAlive>(entity).ok()?;
+                    let keep_alive = state.universe.get_mut::<IncomingKeepAlive>(entity).ok()?;
 
                     if matches!(*conn_state, ConnectionState::Play)
-                        && (current_time - keep_alive.id) >= fifteen_seconds_ms
+                        && (current_time - keep_alive.id) >= 15000
```

Comment:
Could you rename the field `id` to `timestamp` instead? Since I feel like it'd make more sense in the future when looking at this code. Since right now it looks like a quite weird calculation between an id and timestamp :)